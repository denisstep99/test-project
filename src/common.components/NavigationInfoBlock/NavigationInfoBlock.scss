.navigation-info-block {
  &__label {
    font-size: 18px;
    color: #3C6CB5;
  }

  &__bread-crumb {
    padding-right: 4px;
    color: #868686;
  }

  &__bread-crumb:not(:last-child)::after {
    text-decoration:none;
    display:inline-block;
    content: '/';
    margin-left: 4px;
  }

  &__bread-crumbs {
    font-size: 16px;
    padding-top: 8px;
  }
}