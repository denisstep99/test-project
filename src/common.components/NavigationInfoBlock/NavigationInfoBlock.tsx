import './NavigationInfoBlock.scss';
import * as React from "react";
import cx from 'classnames';

import {Link} from "react-router-dom";

interface IBreadCrumb {
    label: string;
    link?: string;
}

interface INavigationInfoBlockProps {
    className?: string;
    /**
     * Текущий раздел
     */
    currentSection: string;
    /**
     * Хлебные крошки к подразделам
     */
    breadCrumbs?: Array<IBreadCrumb>
}

export const NavigationInfoBlock: React.VFC<INavigationInfoBlockProps> = (({
                                                                               className,
                                                                               currentSection,
                                                                               breadCrumbs
                                                                           }) => {
    return (
        <div className={cx('navigation-info-block', className)}>
            <div className="navigation-info-block__label">
                {currentSection}
            </div>
            <div className="navigation-info-block__bread-crumbs">
                {
                    breadCrumbs?.map((breadCrumb, index) => {
                        const {link, label} = breadCrumb;

                        if (link) {
                            return (
                                <Link key={index} to={link} className="navigation-info-block__bread-crumb">
                                    {label}
                                </Link>
                            );
                        }
                        return (
                            <span key={index} className="navigation-info-block__bread-crumb">
                                {label}
                            </span>
                        );
                    })
                }
            </div>
        </div>
    );
});