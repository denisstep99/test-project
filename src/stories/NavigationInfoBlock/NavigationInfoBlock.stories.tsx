import React from 'react';
import StoryRouter from 'storybook-react-router';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import {NavigationInfoBlock} from "../../common.components/NavigationInfoBlock/NavigationInfoBlock";

export default {
    title: 'Example/NavigationInfoBlock',
    component: NavigationInfoBlock,
    argTypes: {},
    decorators: [StoryRouter()]
} as ComponentMeta<typeof NavigationInfoBlock>;

const Template: ComponentStory<typeof NavigationInfoBlock> = (args) => <NavigationInfoBlock {...args} />;

export const Default = Template.bind({});
Default.args = {
    currentSection: "Параметры оператора MVNO",
    breadCrumbs: [
        {label: "Основное"},
        {label: "default operator"}
    ],
};

export const Linked = Template.bind({});
Linked.args = {
    currentSection: "Параметры оператора MVNO",
    breadCrumbs: [
        {label: "Основное", link: "/"},
        {label: "default operator", link: "/2"}
    ],
};