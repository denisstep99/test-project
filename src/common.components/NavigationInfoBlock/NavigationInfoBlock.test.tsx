import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {NavigationInfoBlock} from './NavigationInfoBlock';
import {Link} from "react-router-dom";

describe('NavigationInfoBlock component.', () => {
    let navigationInfoBlock: ShallowWrapper;

    beforeEach(() => {
        navigationInfoBlock = shallow(<NavigationInfoBlock currentSection="Параметры оператора MVNO" breadCrumbs={[
            {label: "Основное"},
            {label: "Побочное", link: "/link_1"},
            {label: "Лишнее", link: "/link_2"},
        ]}/>);
    });

    it('Should render correctly', () => {
        expect(navigationInfoBlock).toMatchSnapshot();
    });

    it('Breadcrumbs with link param should be links', () => {
        const breadCrumbs = navigationInfoBlock.find('.navigation-info-block__bread-crumb');
        expect(breadCrumbs).toHaveLength(3);

        const breadCrumbsLinks = breadCrumbs.filter(Link);
        expect(breadCrumbsLinks).toHaveLength(2);
    });
});