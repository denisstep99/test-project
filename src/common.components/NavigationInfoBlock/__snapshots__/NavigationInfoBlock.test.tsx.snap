// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationInfoBlock component. Should render correctly 1`] = `
<div
  className="navigation-info-block"
>
  <div
    className="navigation-info-block__label"
  >
    Параметры оператора MVNO
  </div>
  <div
    className="navigation-info-block__bread-crumbs"
  >
    <span
      className="navigation-info-block__bread-crumb"
      key="0"
    >
      Основное
    </span>
    <Link
      className="navigation-info-block__bread-crumb"
      key="1"
      to="/link_1"
    >
      Побочное
    </Link>
    <Link
      className="navigation-info-block__bread-crumb"
      key="2"
      to="/link_2"
    >
      Лишнее
    </Link>
  </div>
</div>
`;
